socketio_middleware/src/socketio.rs

```diff
@@ -385,25 +388,39 @@ impl SocketIOWrapper {
 
                             let _ = self.socket.send(Message::Text(content)).await;
                         }
+
                         SocketIOMessage::Join(room_id) => {
-                            self.rooms.push(room_id.to_string());
-                            join_channel_to_room(
-                                &room_id,
-                                ChannelPair::new(&self.sid, self.sender()),
-                            );
+                            // check if room_id exist. Don't use return because of the following process such as PING/PONG.
+                            if false == self.rooms.contains(&room_id) {
```

Comment:
same as above, no need to compare to the boolean, just `if !self.rooms.contains(&room_id) {`

Reply:
changed